### PR TITLE
MM-15336 - Improve profile popover for posts with overwritten usernam…

### PR DESCRIPTION
**Pull request by nadalfederer**
#### Summary

Improve profile popover for posts with overwritten username or icon.
Pass appropriate props to Popover component.
#### Ticket Link

  Fixes https://github.com/mattermost/mattermost-server/issues/13584

#### Screenshots

Screenshot from 2020-01-13 04-19-23

### CODE REVIEW

File: i18n/en.json [hunk 1]
```diff
@@ -3287,6 +3287,7 @@
   "upload_overlay.info": "Drop a file to upload it.",
   "user_list.notFound": "No users found",
   "user_profile.account.editSettings": "Edit Account Settings",
+  "user_profile.account.post_was_created": "this post was created by an integration from",
```
Comment (hanzei) [inline i18n/en.json]: This has to be moved behind `user_profile.account.localTime` to fix the build.

**Comment by jasonblais** [comment]
Hi @aaronrothschild, kind reminder to help with PM review for this PR

cc @DHaussermann for QA review

**Comment by cpanato** [comment]
will remove this test server since this PR is stale

**Comment by jasonblais** [comment]
@aaronrothschild @DHaussermann kind reminder to help with reviews here

**Comment by aaronrothschild** [comment]
@nadalfederer Question: When is this behavior triggered?
When a webhook makes a post, and has a "username override"?

**Comment by nadalfederer** [comment]
>When a webhook makes a post, and has a "username override"
Yes, I have created a custom post with required parameters

**Comment by jasonblais** [comment]
cc @aaronrothschild on above

**Comment by nadalfederer** [comment]
I resolved the conflicts

**Comment by hanzei** [comment]
@DHaussermann Gentle reminder to review this PR. Could another QA reviewer jump on this, if you are overloaded at the moment? Also, please let me know, if you need help with QA testing this PR.

**Comment by DHaussermann** [comment]
@nadalfederer or @hanzei  Can this change be tested on the spin up webapp server above? Unless I'm misunderstanding, I don't see the expected here. Is there a corresponding PR in a different repo that is also needed?

The profile pop-over is still showing that of the webhook creator. I was expecting as in the screen above that the profile would match the username and image used by the webhook for positing.

Thoughts?

**Comment by hanzei** [comment]
@DHaussermann Did you overwrite the user icon or image?

**Comment by DHaussermann** [comment]
@hanzei Yes. I also tried overriding it in the webhook payload as well as with the incoming webhook config.
 From what I understand, this PR should update the profile pop-over when this is being done.

**Comment by hanzei** [comment]
@nadalfederer Could you please help setting up a webhook on the test server above?

**Comment by DHaussermann** [comment]
@hanzei and @nadalfederer Just to make sure my comment was clear, the override is configured and working.
WebHookOveride
I'm confused because I was expecting to see the enhancement in functionality in this case with the pop-over showing these values.

**Comment by hanzei** [comment]
@nadalfederer Could you please take a look?

**Comment by nadalfederer** [comment]
I'll look into it

**Comment by nadalfederer** [comment]
@DHaussermann  Hi, I'm sorry for the late response. 
I see that everything is working as expected. Can you point out what is not working exactly?
Peek 2020-04-17 18-31
This is before and after change
Screenshot from 2020-04-17 18-21-38
Screenshot from 2020-04-17 18-42-31

**Comment by hanzei** [comment]
:man_facepalming: I should have not merged `master` into this branch. Sorry @DHaussermann and @nadalfederer

**Comment by DHaussermann** [comment]
Hi @nadalfederer, Sorry for the delay. Thanks for looking at this. Yes, based on the images you've shared - this is the change I was expecting to see. But, strangely I was seeing the old behavior.

I will check this again shortly 👍

**Comment by DHaussermann** [comment]
@nadalfederer I'm confused as to how I'm not seeing the same thing as you. On the PR server above I'm afraid I'm still seeing the old behavior where the pop-over shows details the sysadmin user.
pop-over

I went to the server in an incognito browser just in case I had something cached. I can't see what the difference is. Maybe I missing something? If you go onto the server without changing config and look  at this post for example https://mattermost-webapp-pr-4663.test.mattermost.cloud/pr4663/pl/hdck3wzefbfm3eddjbz9f8wnyo Do you see the updated username and profile image now?

**Comment by nadalfederer** [comment]
@DHaussermann  I see what's the difference you are clicking on the avatar, and I instead clicking on the username. I missed that, I'll fix it soon

**Comment by nadalfederer** [comment]
@DHaussermann can you check out this? I fixed the current issue

**Comment by DHaussermann** [comment]
- Username and profile image both show updated pop-over
- Pop-over is updated when profile image and username are configured in app
- Pop-over is updated when profile image and username are passed in with payload
- Regression tested that `Enable integrations to override usernames` is still respected

Thanks @nadalfederer this is great!
One last small thing - the text in the bottom of the pop-over says `this post was created by an integration from...`  There is capitalization missing here. It should say `This`

**Comment by nadalfederer** [comment]
It fixed now

**Comment by hanzei** [comment]
@nadalfederer This is such an awesome improvement to webhooks. Great work @nadalfederer :tada:

**Comment by jasonblais** [comment]
Woot :tada: :tada: Thank you @nadalfederer!

**Pull request merged**